### ISSUE TITLE
[Feature request] Show solid background in maximized state instead show background in maximized state

### ISSUE THREAD

**Issue by michaldybczak** [psifidotos/Latte-Dock#713]
Since dynamic colors of fonts based on wallpaper is not realistic at the moment and may be vision for future, I have different solution for the crappy visibility of white fonts on most wallpapers with transparent panel.

At the moment latte allows for theme color on a panel with maximized windows. In result, when we use semi-transparent theme, we get semi-transparent panel, which looks not particularly good in maximized mode. Here are examples.
1. Solid panel:
https://puu.sh/xAWDs/d622850400.png
2. Semi-transparent panel:
https://puu.sh/xAWUZ/49b7bb0278.png

Those are details, but in second example, coherent screen look is no more and will change dependently on wallpaper. This is because at this moment we can choose a semi-transparent panel which fixes issue of dark fonts visibility but we can't make it solid on maximize without being solid in un-maximized state. So either we have semi-transparent panel all the time or transparent with maximized solid.

The idea is to provide a third option: semi-transparent in un-maximized state, solid in maximized state.

It can be for example achieved in this way:

**Instead "show background only for maximized windows" we would have "show solid background only for maximized windows"**
This way, we could choose panel color by changing a theme and then tweak it with opacity option like it is now, so it could be fully transparent or semi-transparent as it is now, but on maximizing we would get solid color which may suit general window color, achieving nice coherent design.

At the moment we already have full control over un-maximized state by choosing theme, shadow/solid, opacity options so we can make any panel as we want, but we have no option to choose it in maximized state. If we mark "show background only for maximized windows" it will simply keep the same settings so if I tweak it to full transparency, I will have transparent panel also in maximized state and the tweak will have no visible effect. If I do semi-transparent panel, the same, no visible effect.

If you follow me, **I propose to turn around how effects are applied. Instead creating forceful transparency in un-maximized state - which is what current option does, it would be better to create forceful solid background in maximized state.** It would allow for more control and more possibilities of various settings. 

So a user could easily turn off the panel background and have transparent panel all the time or in un-maximized state. Or tweak it to semi-transparency and have it all the time or only in un-maximized state (solid in maximized). The only con I see, there would be no option to change from solid/semi-transparent panel in un-maximized to transparent in maximized, but I just can't imagine this scenario would be useful in any way.

Also, I think it would make the whole Latte UI more intuitive. Now in look section you can change panel's look all the way from full transparency to solid background. This is really awesome. Current tweak nullifies those settings in un-maximized state. So we have only control over maximized state but not un-maximized, while in maximized, only solid background has point while in un-maximized you may need various settings, so the other way around. The feature I request would provide that.

Just think it through, I may be wrong. Others are welcome to add their thought on it. So the basic question is: is it better to have full control over panel's look in maximized or un-maximized state?

I'd say in un-maximized, while in maximized only solid background is useful.

**Comment by psifidotos** [comment]
In version 0.7.1, we currently provide the shadow customizability to be used when the dock is fully transparent in mode (solid background only for maximized windows)...

the idea is good it could be just one more option in Tweaks page, "show solid background only for maximized windows" and the settings for the panel background to be applied only in non-maximized state... One thing is of course the panel shadows option which must be thought... how it is going to be used in the new state and if we want one more option for this... 

Either way new features that add new translation strings are going to be added only in version 0.8, so we can open also a thread in kde forums and ask the VDG option also how all these options could be provided better and how for v0.8

**Comment by michaldybczak** [comment]
Ah, didn't notice the new "hide panel shadow in maximized windows". That's an awesome option. I had to manually modify a theme to get rid of the shadow which looks truly awful. I called my modified theme "MacBreeze Shadowless". I haven't uploaded it to the KDE Store yet. It wasn't easy to get rid of that shadow so having a tweak to do it for you is a life saver for users ;). We could use any theme now :D. Thanks!

As to adding my idea as a new option, I guess it can be good. I thought about proposing that but... I am aware of complications it may cause. It's typical for KDE tou have so many options that some will contradict each other or not work in certain setups. You could see that in AWC widget where many settings can produce rubbish results which confuses user. I spend many hours testing various settings till I found working ones... Not all users are ready to to that. Most will react: "this doesn't work" or "this works badly" and will blame software or developer, while this may be just a mismatch of options. And I am sure you don't want that "bug reports" that aren't real bugs either.

It's really hard to design system and UI in a way it would have tons of options and yet provide coherent and intuitive results without mismatches. Some go for automatic switch off of options if one contradicts other (compiz), but that can be confusing as hell too. I know I didn't like that part of compiz. Also kwin has so many options that seem "not to work" so we have no idea what they actually are...

Anyway, you may add this new option and test it. Then publish it and if some additional issues arise, we will give you feedback and then you decide what to do: redesign it, tweak it further, etc. This is anyway work in progress so it's hard to expect perfect results. It's astonishing how well latte works and how much value it adds to the Plasma and how quickly it happen. Not so long ago, I never know Latte existed... and now I can't imagine Plasma without it :D.

**Comment by psifidotos** [comment]
We will find a way to support it correctly . :-) Our approach is that when adding new options - features we try to protect the users from multiple options, we try somehow to provide sane results.

Just to add in your thought that Latte is just nine months old

**Comment by michaldybczak** [comment]
Wow! You developed it insanely fast and you got recognized also quickly.

Thanks for hearing me out. I will finally be able to use any wallpaper I want, not just the ones with extreme bright upper area. I like the look of dark themes and windows but I can't force myself to use them,they're hard to read and they hurt my eyes (ironically the same reasons are mentioned why people can't use bright themes/windows... ;P) so I prefer light themed windows, while having transparent or semi-transparent panel (if fonts/icons looks illegibly on it).

Perfect situation would to have always transparent panel with always readable fonts, but it would require my initial idea to be implemented, dynamic font colors, just like it is on Android, but we're not there yet. Maybe someday ;).

**Comment by psifidotos** [comment]
ok... lets have some facts... I combined this with #702 these are some screenshots...

![image](https://user-images.githubusercontent.com/2216474/30552751-b9c91e86-9ca7-11e7-918e-a11dbac42e00.png)

![image](https://user-images.githubusercontent.com/2216474/30552769-cc6523b4-9ca7-11e7-9b51-5f4f32ad7a44.png)

![image](https://user-images.githubusercontent.com/2216474/30552827-00cc9ede-9ca8-11e7-8ed2-596c8387745b.png)

![image](https://user-images.githubusercontent.com/2216474/30552846-1134c8dc-9ca8-11e7-93ec-aa27ee241aba.png)


do you have a screenshot to send me to test how it would look?

**Comment by psifidotos** [comment]
and these...

![image](https://user-images.githubusercontent.com/2216474/30553372-b7fef5e2-9ca9-11e7-9e30-828c3aa95484.png)
![image](https://user-images.githubusercontent.com/2216474/30553418-d0d561dc-9ca9-11e7-9076-da6f606fd873.png)

**Comment by michaldybczak** [comment]
Can you switch to light panel and show how it looks like? Those examples are nice but not the effect I am going for.
The whole screen area should be stylish and coherent in maximized mode. This is how mac is designed. Every detail matters. It's hard to achieve such level of style on linux desktop but Plasma is very close to it. And of course, it still gives more freedom then mac ;D.

**Comment by psifidotos** [comment]
I think it cant be better from what you have sent... your walllpaper is chosen in that way in order the black font to provide the best contrast and the sky background is clear enough in order for the elements to be distinguished correctly...

maybe I didnt understand, what would you improve from your screenshots?

**Comment by michaldybczak** [comment]
In most wallpapers black fonts become almost invisible. So my options would be:

1) have a wallpaper like the one above, then I can set panel transparent and it would get solid on maximize (that works now, great)
2) have a wallpaper that doesn't fit well with black fonts (most wallpapers out there) so I would have to set semi-transparency for the panel and it would still turn solid on maximize - that scenario is not possible now, because in order to have semi-transparency, I have to turn off the tweak with solid background on maximize, so I have semi-transparency all the time, which on maximize is not a good idea.

So to make it short, I would like to have the same effect (solid background on maximize) with full transparency and semi transparency (which I set in Look area) if wallpaper doesn't fit the fonts. That is why I was proposing to change the tweak into **Show solid background in maximized state instead show background in maximized state**.

You can test it yourself.
1. Just choose a light panel with black icons (hard to find, breeze light should look like that, but from whatever reason it has two font colors, not sure if that's bug on my system or it is just a messed theme).
2. Make assumption that you need solid panel on maximize.

Then switch wallpapers and you will see that only very few of them are workable. So the solution to them would be to add some opacity, but then solid background on maximize tweak must be turned off... which  I don't want. So there is no way to have semi transparency and solid background on maximize at the same time.

My current possible options are: have transparent panel and solid on maximize or have solid all the time or semi transparent all the time.

Basically, your tweak unnecessarily is tied with transparency, because you can set it in Look section manually. If you would just have "solid on maximize" independently on the panel, then a user could set panel they desire (full transparency or various stages of semi-transparency) and then choose solid on maximize. The only con of that solution would be: if someone has solid panel, won't see any effect. But I have the same with shadow, I don't have it on panel, so turning "no shadow on panel" does nothing, the same with blur if there is full transparency on, no blur visible. So like you see, some combination of options already won't produce a visible effect anyway. It's not that bad. You may add some hints in brackets, like *visible when panel opacity is lower then 100%.

**Comment by psifidotos** [comment]
can you send some screenshots proving that using a light theme and setting a panel transparency improves the contrast and readability?

**Comment by michaldybczak** [comment]
Here is some standard wallpaper with too dark background and fonts are hard to read:
![screenshot_20170919_182503](https://user-images.githubusercontent.com/10132958/30603473-dd2dbee8-9d67-11e7-9bed-e8c29cc32921.png)

With semi transparent panel (80% opacity):
![screenshot_20170919_182654](https://user-images.githubusercontent.com/10132958/30603666-82021306-9d68-11e7-8328-49d5731df64e.png)

In maximized mode (when background is so homogeneous, it doesn't look that bad):
![screenshot_20170919_183156](https://user-images.githubusercontent.com/10132958/30603765-d4924f3c-9d68-11e7-9dd8-b236c4a11f98.png)

It would be nice to still have the full control over panel look with the solid on maximize effect. Then we could have everything. Now solid on maximized is tied only to full transparency (which is nice but limiting).

**Comment by psifidotos** [comment]
By playing with this for more than two weeks I think that this will be the default behavior concerning panel solidness in maximized windows and panel transparency for normal windows.

the user settings will influence only the panel transparency for normal windows, for maximized windows no transparency will be allowed...

having user setting panel transparency for maximized windows doesnt make much sense I think (this is the default behavior for v0.7)

**Comment by michaldybczak** [comment]
OK, will gladly see how it works. If there will be problems, I will let you know.
I assume, that not enabling tweak will still allow us to decide how panel looks like, even with full transparancy, althought this is completely useless? Just for the sake of choice ;).

**Comment by psifidotos** [comment]
The user will be able to set everything for the panel appearance when there isnt a maximized window

**Comment by michaldybczak** [comment]
OK, will see how it works. I like that I am able to trigger this dynamic panel or leave it as it is (customizable by opacity and visibility), but maybe I won't miss it. We'll see.

**Comment by psifidotos** [comment]
> OK, will see how it works. I like that I am able to trigger this dynamic panel or leave it as it is (customizable by opacity and visibility), but maybe I won't miss it. We'll see.

you mean that you like having transparency for your top panel when there is a maximized window?

**Comment by michaldybczak** [comment]
Not really. I used transparent panel for a long time and, although effect is cool, it's often unpractical and hard to read. However, I can imagine some wallpaper and settings that it might be ok. I'm for all possibilities, because there will be always the one person or the one setting that will make sense.

I prefer transparent panel unmaximized, but solid in maximized state, but I just want to have also the option to have semi-transparent panel in unmaximized state, for some wallpapers that are too dark for white fonts/icons (most wallpapers out there). So for both cases, solid background in maximized state. Still, some may not need such coherent look and some semi-transparent panels or certain wallpapers may be ok, also in maximized state, so I wouldn't want to get rid of that possibility.
